### PR TITLE
local docker-compose development

### PR DESCRIPTION
**Pull request by gkostin1966**
see README.md

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,114 +1,96 @@
...
+      context: .
     ports:
-      - "8983:8983"
+      - "3000:3000" # Rails
+      - "1234:1234" # RubyMine
+      - "26162:26162" # RubyMine
     environment:
-      - SOLR_JAVA_MEM=-Xms1024m -Xmx1024m
-      - SOLR_HEAP=1024m
+      - RAILS_ENV=development
```
Comment (respinos) [inline docker-compose.yml:17]: Could you add the analytics environment settings?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,114 +1,96 @@
...
+      - REDIS_URL=redis://redis:6379
+      - DATABASE_URL=postgresql://postgres:postgres@db/umich-arclight-development
+      - SOLR_URL=http://solr:8983/solr/umich-arclight-development
     volumes:
-#      - ./solr/arclight:/var/solr/data/arclight
-      - ./solr/arclight/conf:/solr_config:ro
+      - .:/opt/app
+      - ./sample-ead:/data:ro
```
Comment (respinos) [inline docker-compose.yml:23]: If you're mirroring production, `/data` cannot be `ro`; the PDF workflow writes to `$FINDING_AID_DATA/pdf`.

Although that's missing too?

File: docker-compose.yml (hunk 1 above)
Comment (gkostin1966) [reply docker-compose.yml:17]: Okay, the Google stuff, yes.

File: docker-compose.yml (hunk 2 above)
Comment (gkostin1966) [reply docker-compose.yml:23]: The default FINDING_AID_DATA is /data but you need the environment variable, I understand.

Comment (respinos) [reply docker-compose.yml:23]: The default is fine, but I need it rw

File: docker-compose.yml (hunk 1 above)
Comment (gkostin1966) [reply docker-compose.yml:17]:
```
    environment:
      - RAILS_ENV=development
      - FINDING_AID_DATA=/var/opt/app/data
      - GOOGLE_ANALYTICS_DEBUG=false
      - GOOGLE_ANALYTICS_TRACKING_ID=0
      - DATABASE_URL=postgresql://postgres:postgres@db/umich-arclight-development
      - SOLR_URL=http://solr:8983/solr/umich-arclight-development
      - REDIS_URL=redis://redis:6379
```

File: docker-compose.yml (hunk 2 above)
Comment (gkostin1966) [reply docker-compose.yml]: Declaration
```
services:
  app:
    volumes:
      - data:/var/opt/app/data
```
Initialization
```
volumes:
  data:
    driver_opts:
      type: none
      device: ${PWD}/sample-ead
      o: bind
```

Comment (gkostin1966) [reply docker-compose.yml]: Changed my mind and decide NOT to hack the "data" volume to do automagically sync with the local `./sample-ead` directory. 

```
volumes:
    data:
```

**Review by respinos — APPROVED** [review]
Works! I would love to know why this is different from the scheme we've been using, but I'm not sure I fully grok _that_ version either, so ¯\_(ツ)_/¯

**Comment by gkostin1966** [comment]
It is the same but different. Basically I want to run test/spec locally which requires a test database and a test solr core. Also I added more volumes to be a bit more persistent so things don't missing when I docker-compose down then and up again. Also there are like 50+ pending specs that I would like to review, understand, and get working again.

**Pull request merged**